pkg/kubernetes/scripts/registry.js

```diff
@@ -163,6 +164,59 @@
                 return select().kind("Project");
             };
 
+            // TODO: invalidate cache on role changes
+            var sarCache = { };
+
+            $scope.subjectAccessReview = function subjectAccessReview(project, user, verb, resource) {
+                var result = false;
+
+                // if no project was given, check the first one as an approximation
+                // TODO: is that sufficient, or should we iterate over all
+                // projects? (much more expensive)
+                if (user && !project) {
+                    var projects = select().kind("Project");
+                    for (var p in projects) {
+                        project = projects[p].metadata.name;
+                        break;
+                    }
+                    if (!project)
+                        // we didn't init sufficiently far yet, defer to the
+                        // next call; or the user really has no projects
+                        return false;
+                }
+
+                var key = project + ':' + (user ? user.metadata.name : "")+ ':' + verb + ':' + resource;
+
+                // cached already?
+                if (key in sarCache) {
+                    result = sarCache[key];
+                    console.log("XXX subjectAccessReview", key, ": cached value", result);
+                    return result;
+                }
+
+                // pre-fill cache with "false" to avoid multiple concurrent requests
+                sarCache[key] = false;
+
+                var request = {
+                    kind: "SubjectAccessReview",
+                    apiVersion: "v1",
+                    namespace: project,
+                    verb: verb,
+                    resource: resource
+                };
+                console.log("subjectAccessReview: doing SubjectAccessReview call for", project, verb, resource);
+                // TODO: use OPENSHIFT constant from kube-client.js?
+                methods.post("/oapi/v1/subjectaccessreviews", request)
```

Comment:
If you add subjectaccessreview to SCHEMA in kube-client.js you should be able to have resourcePath build the path for you.